source/operators/captions/webcaptions.py

```diff
@@ -0,0 +1,1016 @@
+
+    code_lookup = {
+        "ar":"arb",
+        "zh":"cmn-CN",
+        "zh-TW":"cmn-CN",
+        "da":"da-DK",
+        "de":"de-DE",
+        "en":"en-GB",
+        "es":"es-ES",
+        "es-MX":"es-MX",
+        "fr":"fr-FR",
+        "fr-CA":"fr-CA",
+        "it":"it-IT",
+        "ja":"ja-JP",
+        "hi":"hi-IN",
+        "ko":"ko-KR",
+        "no":"nb-NO",
+        "nl":"nl-NL",
+        "pl":"pl-PL",
+        "pt":"pt-PT",
+        "ro":"ro-RO",
+        "ru":"ru-RU",
+        "sv":"sv-SE",
+        "tr":"tr-TR"
+    }
+
+    if translate_language_code in code_lookup:
+        return code_lookup[translate_language_code]
+    else:
+        return "not supported"
+
+# Create a Polly track using the text only transcript for each language in the WebCaptions collection for this AssetId
+# The pacing of this polly track is determined bu the Polly service
+def start_polly_webcaptions (event, context):
+    print("We got this event:\n", event)
+
+    operator_object = MediaInsightsOperationHelper(event)
+    webcaptions_object = WebCaptions(operator_object)
+
+    captions_collection = webcaptions_object.GetWebCaptionsCollection()
+    print("INPUT CAPTIONS COLLECTION")
+    print(json.dumps(captions_collection))
+
+    for caption in captions_collection:
+
+
+        # Always start from WebCaptions data since these are the most recently edited version
+        # Convert WebCaptions to a text only transcript
+        transcript = webcaptions_object.GetTextOnlyTranscript(caption["TargetLanguageCode"])
+
+
+        # If input text is empty then we're done.
+        if len(transcript) < 1:
+            operator_object.update_workflow_status("Complete")
+            return operator_object.return_output_object()
+
+        # Get language code of the transcript, we should just pass this along in the event later
+        language_code = translate_to_polly_language_code(caption["TargetLanguageCode"])
+
+        if language_code == "not supported":
+            caption["PollyStatus"] = "not supported"
+        else:
+
+            try:
+                # set voice_id based on language
+                response = polly.describe_voices(
+                    #Engine='standard'|'neural',
+                    LanguageCode=language_code
+                    #IncludeAdditionalLanguageCodes=True|False,
+                    #NextToken='string'
+                )
+
+            except Exception as e:
+                operator_object.update_workflow_status("Error")
+                operator_object.add_workflow_metadata(PollyCollectionError="Unable to get response from polly describe_voices: {e}".format(e=str(e)))
+                raise MasExecutionError(operator_object.return_output_object())
+            else:
+                # just take the fisrt voice in the list.  Maybe later we can extend to choose voice based on other criteria such
+                # as gender
+                if len(response["Voices"]) > 0 :
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`.

 [Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)